policies/coding-style.md

```diff
@@ -591,21 +591,89 @@ should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
-## Chapter 15: Miscellaneous
+## Chapter 15: Expressions
 
-Do not use `!` to check if a pointer is NULL, or to see if a str...cmp
-function found a match.  For example, these are wrong:
+Avoid needless parentheses.
+For example, do not write
+```c
+    if ((p == NULL) && (!f(((2 * x) + y) == (z++))))
+```
+but
+```c
+    if (p == NULL && !f(2 * x + y == z++)).
+```
+
+For clarity, always put parentheses when mixing `&&` and `||` operations.
+For example,
+```c
+    if ((a && b) || c)
+```
+
+Put parentheses around macro bodies that can be used as expressions.
 
+Put parentheses around uses of macro arguments
+(unless they are passed on as-is to a further macro or function).
+For example,
 ```c
-    if (!(p = BN_new())) ...
-    if (!strcmp(a, "FOO")) ...
+#define SQUARE(a) ((a) * (a))
```

Comment:
Not sure if the macro section would be a better place for these?

Reply:
Good point that there is an overlap with the macro section.
Yet maybe some redundancy is even good in the sense that some reader may only have a look at either of these two place when looking for guidance for macro expressions?
And pragmatically, any further change to this PR would invalidate the approvals already given.

Reply:
I agree, this is about macros rather than about expressions.

> any further change to this PR would invalidate the approvals already given.

This is an example of where the rules (two weeks quarantine, vote, etc.) are not helping but preventing improvements.

If there's a new rule about trivial changes that I've heard mentioning, then perhaps in the future, moving around and reformulating information (without changing the essence of it) can be considered trivial and done without the full process... (I hope so.)

Reply:
Here is a suggestion for a pragmatic way out in this case:
I anyway plan to provide further chunks, extending this document with further items of https://github.com/openssl/openssl/issues/10725#issue-543528689.
As far as requested, I can simply sort out overlaps between the two sections with the next PR.